Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationDatabase.cs

```diff
@@ -119,6 +128,36 @@ public void ConfigureGarbageCollection(bool shouldDoGc)
             }
         }
 
+        private void ConfigureInMemoryDatabaseCache(bool isDatabaseWritable)
+        {
+            if (_configuration.CacheEnabled)
+            {
+                // This clear is actually safe, as no operations should happen concurrently with this function.
+                _inMemoryCache.UnsafeClear();
+
+                Interlocked.Exchange(ref _cacheUpdatesSinceLastFlush, 0);
+
+                lock (_cacheFlushTimerLock)
+                {
+                    IsInMemoryCacheEnabled = isDatabaseWritable;

```

Comment:
>IsInMemoryCacheEnabled [](start = 20, length = 22)

Should this be set to false if _configuration.CacheEnabled == false

Reply:
This branch can only be taken if _configuration.CacheEnabled is true in the first place. IsInMemoryCacheEnabled is initialized to false by default

---
In reply to: [291336968](https://github.com/microsoft/BuildXL/pull/403#discussion_r291336968) [](ancestors = 291336968)